src/shell/shell_cmds.cpp

```diff
@@ -482,6 +482,20 @@ static std::string to_search_pattern(const char *arg)
 	return pattern;
 }
 
+// Map a vector of file listing results to a vector of field widths.
+static std::vector<int> map_name_lengths(const std::vector<DtaResult> &results,
+                                         int padding)
+{
+	std::vector<int> ws;
+	ws.reserve(results.size());
+	for (const auto &entry : results) {
+		const int len = static_cast<int>(
+		        strnlen(entry.name, sizeof(entry.name)));
+		ws.push_back(len + padding);
+	}
+	return ws;
+}
```

Comment:
I think the names in here are making it harder for me to review; I'll just mention how the names are messing w/ me, and then we can try a round 2 :-) 

 - Using the word "map" in the description and func name has me thinking we're making some association between A and B (like a real std::map, or something that represents that).  Suggest not using 'map', especially if we're just gathering lengths.

- The variable "results" has me confused that this is where we're storing the results.. but instead, it's actually an input to this function (and a const one, at that).  Incoming variables should be named relative to what they're providing the function with.  Is "results" the list of filenames we're going to process? Or maybe "file entries" is more technically correct, given they're not strings and instead DTAs (whatever a DTA is).

- The comment mentions "to a vector of field widths":  What's a field? Is it actually just the filename length? (if so, suggest not using the "field" word). I'm looking for a variable that's storing these field width.. but I'm only seeing `ws`:  I'm assuming the "w" stand for width? What does the "s" stand for? Suggest  renaming "ws", but preferably something other that "field_widths", because I would still be confused by "field" :-)

The logic looks small and simple, so hopefully round-2 w/ new names and description will get me over this hurdle.

Reply:
Word `map` - ah, my functional programming bias is showing :) - I used it in [this meaning](https://en.wikipedia.org/wiki/Map_(higher-order_function)). Usuall C++ parlance uses `transform` word instead…

Variable `results` - you're right - I'll fix this.

`field widths` - I started with `filename widths`, but then figured out that these might be directories as well (and we modify them with padding), so after several iterations I arrived at `fields` word… and then replaced it in most contexts with `cells` (as in table). I'll try to find better naming for this, the names I picked up are too weird.